packages/backend/src/workers/provider/InferenceProvider.ts

```diff
@@ -46,6 +46,9 @@ export abstract class InferenceProvider implements IWorker<InferenceServerConfig
   }
 
   abstract enabled(): boolean;
+  prePerform(_config: InferenceServerConfig): Promise<void> {
+    return Promise.resolve();
+  }
```

Comment:
not sure to understand the rational behind the `prePerform` ? why not having just a protected function in the openvino provider, and just doing the same `prePerform` logic in `perform`, before doing anything else ?